app/models/user.rb

```diff
@@ -13,6 +13,12 @@ class User < ApplicationRecord
   has_many :pairing_users, foreign_key: :pairee_id, class_name: 'Pairing'
   has_many :pairers, through: :pairing_users
 
+  has_many :rock_users, foreign_key: :pebble_id, class_name: 'RockAndPebble'
+  has_many :rocks, through: :rock_users
+
+  has_many :pebble_users, foreign_key: :rock_id, class_name: 'RockAndPebble'
+  has_many :pebbles, through: :pebble_users
```

Comment:
Great work on this! I had to go do some reading to remember how these work to even review this. 😅 

I see that you were following how we set up the Pairing self-referential association; our terminology ended up biting us later, and I think we could try to make the terminology clearer this time around. What do you (and others) think of changing `rock_users` to `pebble_roles` and `pebble_users` to `rock_roles` for clarity? 

(If I understand correctly, `rock_users` is looking up the rows on the `rock_and_pebbles` table where the user in question was a pebble, so that their corresponding "rocks" can be found (line 17); and vice versa with lines 19-20.)